client/src/TagExplorer/TagExplorerComponent.js

```diff
@@ -0,0 +1,318 @@
+    });
+    this.debounced_set_query(new_query);
+  }
+  debounced_set_query(new_query) {
+    this.props.set_query(new_query);
+    this.timedOutStateChange = setTimeout(() => {
+      this.setState({
+        loading: false,
+      });
+    }, 500);
+  }
+  componentWillUnmount() {
+    !_.isUndefined(this.debounced_set_query) &&
+      this.debounced_set_query.cancel();
+    !_.isUndefined(this.timedOutStateChange) &&
+      clearTimeout(this.timedOutStateChange);
+  }
+  clearQuery() {
+    this.setState({ _query: "" });
+    this.props.clear_query("");
+  }
+  render() {
+    const {
+      flat_nodes,
+      is_filtering,
+
+      set_query,
+      toggle_node,
+
+      //scheme props
+      hierarchy_scheme,
+      is_descending,
+      sort_col,
+      col_click,
+      year,
+    } = this.props;
+
+    const explorer_config = {
+      column_defs: get_col_defs({ year }),
+      onClickExpand: (id) => toggle_node(id),
+      is_sortable: true,
+      zebra_stripe: true,
+      get_non_col_content: render_non_col_content,
+      children_grouper,
+      col_click,
+    };
+
+    const { loading } = this.state;
+
+    const root = get_root(flat_nodes);
+
+    const all_category_props = _.map(
+      hierarchy_scheme_configs,
+      ({ id, title, text, is_m2m }) => ({
+        id,
+        title,
+        text,
+        is_m2m,
+        active: id === hierarchy_scheme,
+      })
+    );
+    const current_category = _.find(
+      all_category_props,
+      (props) => props.active
+    );
+
+    const inner_content = (
+      <div>
+        <div style={{ marginTop: "15px" }}>
+          <form
+            style={{ marginBottom: "15px" }}
+            onSubmit={(evt) => {
+              evt.preventDefault();
+              evt.stopPropagation();
+              set_query(evt.target.querySelector("input").value);
+              this.refs.focus_mount.focus();
+            }}
+          >
+            <input
+              aria-label={text_maker("explorer_search_is_optional")}
+              className="form-control input-lg"
+              type="text"
+              style={{ width: "100%" }}
+              placeholder={text_maker("everything_search_placeholder")}
+              onChange={(evt) => this.handleQueryChange(evt.target.value)}
+            />
+            {window.is_a11y_mode && (
+              <input
+                type="submit"
+                name="search"
+                value={text_maker("explorer_search")}
+              />
+            )}
+          </form>
+        </div>
+        <div
+          tabIndex={-1}
+          className="explorer-focus-mount"
+          ref="focus_mount"
+          style={{ position: "relative" }}
+          aria-label={text_maker("explorer_focus_mount")}
+        >
+          {loading && (
+            <div className="loading-overlay">
+              <div style={{ height: "200px", position: "relative" }}>
+                <SpinnerWrapper config_name={"sub_route"} />
```

Comment:
Not imported, is the linter not catching this somehow? Noticed while testing in-app, searching in the tag explorer crashed me.

Reply:
Neither the build output nor vscode pointed this out, strange